harness/determined/horovod.py

```diff
@@ -85,11 +85,15 @@ def cross_size(self) -> Any:
 hvd = _PolyHorovod()
 
 
-def create_hostlist_arg(num_proc_per_machine: int, ip_addresses: List[str]) -> str:
+def create_hostlist_arg(num_proc_per_machine: List[int], ip_addresses: List[str]) -> str:
```

Comment:
can you rename `num_proc_per_machine` to be `slot_counts` or `container_slot_counts` or `per_node_slot_counts` or something?  I dislike that it used to mean the same thing as torch.distributed.launch's `--num_procs_per_node` but now means something entirely different.

I also think that the current name reflects the old meaning well, but is a bit confusing wit the new meaning.